File: java/src/main/java/fr/sdecout/handson/persistence/library/AddressEntity.java
package fr.sdecout.handson.persistence.library;

import fr.sdecout.handson.rest.shared.AddressField;
import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;

@Embeddable
public class AddressEntity {

    @Column(name = "ADDRESS_LINE_1", nullable = false)
    public String line1;

    @Column(name = "ADDRESS_LINE_2")
    public String line2;

    @Column(nullable = false)
    public String postalCode;

    @Column(nullable = false)
    public String city;

    public static AddressEntity from(AddressField address) {
        var instance = new AddressEntity();
        instance.line1 = address.line1();
        instance.line2 = address.line2();
        instance.postalCode = address.postalCode();
        instance.city = address.city();
        return instance;
    }

    public AddressField toAddressField() {
        return new AddressField(line1, line2, postalCode, city);
    }

}

File: kotlin/src/main/kotlin/fr/sdecout/handson/persistence/library/AddressEntity.kt
package fr.sdecout.handson.persistence.library

import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Deprecated("Replace with jOOQ")
@Embeddable
data class AddressEntity(
    @Column(name = "ADDRESS_LINE_1", nullable = false) val line1: String,
    @Column(name = "ADDRESS_LINE_2") val line2: String? = null,
    @Column(nullable = false) val postalCode: String,
    @Column(nullable = false) val city: String,
)
